kernel/environ.ml

```diff
@@ -544,6 +544,14 @@ let set_allow_sprop b env =
 let sprop_allowed env = env.env_typing_flags.sprop_allowed
 
 let set_rewrite_rules_allowed b env =
+  (* We need to be safe with reduction machines *)
+  let flags = typing_flags env in
+  let env = set_typing_flags
+    { flags with
+      enable_VM = flags.enable_VM && not b;
+      enable_native_compiler = flags.enable_native_compiler && not b }
+    env
```

Comment:
I think this should be done in set_typing_flags, not in the convenience wrapper set_rewrite_rules_allowed.
set_typing_flags is the one that actually enforces invariants.

Reply:
I don't understand what you mean, especially as `rewrite_rules_allowed` is not a typing flag but a "regular environment flag"

Reply:
oh right I got confused